protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,253 @@
+pragma solidity ^0.8.20;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (templegold/SpiceAuction.sol)
+
+
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ISpiceAuction } from "contracts/interfaces/templegold/ISpiceAuction.sol";
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+import { AuctionBase } from "contracts/templegold//AuctionBase.sol";
+import { TempleMath } from "contracts/common/TempleMath.sol";
+
+/** 
+ * @title SpiceAuction
+ * @notice Temple Gold is deposited into this contract to bid on a share of distributed Spice token, 
+ *         or the other way round, for an epoch. So Temple Gold can be the bid or auction token at any auction epoch.
+ *         Reward tokens acquired in past epochs can always be claimed. Once bidding for an epoch has ended, users cannot
+ *         claim their bid token and can only claim their share of reward token for epoch.
+ *         Bid and auction tokens could change per auction. These are set in `AuctionConfig`. 
+ *         Config is set before the next auction starts.
+ */
+contract SpiceAuction is ISpiceAuction, AuctionBase {
+    using SafeERC20 for IERC20;
+    using TempleMath for uint256;
+
+    /// @notice Spice auction contracts are set up for 2 tokens. Either can be bid or sell token for a given auction
+    /// @notice uint(TOKEN_A) < uint(TOKEN_B)
+    address public immutable override spiceToken;
+    /// @notice Temple GOLD
+    address public immutable override templeGold;
```

Comment:
`templeGold` and `spiceToken` are also public getters for these two tokens, so I disagree `templeGold` is not required as a separate variable.
Setting `spiceToken = templeGold` is also confusing to me given argument above, unless I'm misunderstanding. happy to be corrected if otherwise

Reply:
I misread the ternary code below, can ignore this